### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by felixodette**
- Create model classes for all entities as shown in the ERD diagram.
- Set up associations between models.

## Use models to insert data
- Familiarize yourself with 'rails c' command.
- Open rails console for your project.
- Create at least one user by running the following code:
   `first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')`
- Create at least 4 posts written by one of the users you created by running the following code:
  `first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
Create at least 6 posts comments for one of the posts you created by running the following code:
   Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )`
- Use other CRUD methods to find, update and delete entities.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (danmainah) [inline app/models/user.rb]: - [ ] @felixodette great work kindly add the foreign key associating the user with posts comments and likes.
`hint`
```suggestion
  has_many :posts, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
+  has_many :likes
+  has_many :comments
+
```
Comment (danmainah) [inline app/models/post.rb:5]: - [ ] Kindly add a method that updates the posts counter for a user.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @felixodette , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by felixodette** [comment]
Hey @danmainah thanks for the review. Added the required code.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes, foreign_key: 'author_id'
```
Comment (AnselemOdims) [inline app/models/user.rb]: - [x] Good job on including the foreign key for the`likes`. Please also add it to the `comments` and `posts` as recommended by the previous reviewer.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @felixodette , 
Good job implementing some of the changes requested by the previous reviewer.

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/user.rb (hunk 3 above)
Comment (felixodette) [reply app/models/user.rb]: Done. Thanks

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @felixodette, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You have done a great job working on the requested changes✔️✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**